packages/core/lib/configureGanacheOptions.js

```diff
@@ -0,0 +1,48 @@
+function sanitizeGanacheOptions(ganacheOptions) {
+  const network_id = ganacheOptions.network_id;
+
+  // Use default network_id if "*" is defined in config
+  if (network_id === "*") {
+    return { ...ganacheOptions, network_id: Date.now() };
+  }
+
+  const parsedNetworkId = parseInt(network_id, 10);
+  if (isNaN(parsedNetworkId)) {
+    const error =
+      `The network id specified in the truffle config ` +
+      `(${network_id}) is not valid. Please properly configure the network id as an integer value.`;
+    throw new Error(error);
+  }
+  return { ...ganacheOptions, network_id: parsedNetworkId };
+}
+
+function configureManagedGanache(config, networkConfig, mnemonic) {
+  const ganacheOptions = {
+    host: networkConfig.host || "127.0.0.1", // Default host for managed ganache
+    port: networkConfig.port || 9545, // Default port for managed ganache
+    network_id: networkConfig.network_id || Date.now(),
```

Comment:
~~Hm, it looks to me like this doesn't sanitize out the case of `network_id = "*"`... that has to be handled, doesn't it?  Or did I just miss where that happens?~~ Nevermind, you handled that separately!  Wondering if it makes sense to combine those?  The way you have it is fine though I think.